beacon/beacon-error.js

```diff
@@ -0,0 +1,30 @@
+"use strict";
+
+if (!this.document) {
+    importScripts("/resources/testharness.js");
+    importScripts("/common/utils.js");
+    importScripts("beacon-common.js?pipe=sub");
+}
+
+test(function() {
+    // Payload that should cause sendBeacon to return false because it exceeds the maximum payload size.
+    var exceedPayload = Array(maxPayloadSize + 1).fill('z').join("");
+
+    var success = navigator.sendBeacon("http://doesnotmatter", exceedPayload);
+    assert_false(success, "calling 'navigator.sendBeacon()' with payload size exceeding the maximum size must fail");
+}, "Verify calling 'navigator.sendBeacon()' with a large payload returns 'false'.");
+
+test(function() {
+    var invalidUrl = "http://invalid:url";
+    // http://osgvsowi/7877590 - Fetch consistently throws TypeMismatchError instead of TypeError
+    assert_throws("TypeMismatchError", function() { navigator.sendBeacon(invalidUrl, smallPayload); },
```

Comment:
Per spec...

> Set parsedUrl to the result of the URL parser steps with url and base. If the algorithm returns an error, or if parsedUrl's scheme is not "http" or "https", throw a "SyntaxError" exception and terminate these steps.

- Chrome throws `SyntaxError`
- Firefox throws `URLMismatchError`
- Edge throws a `TypeMismatchError`

Not sure what to do about this one. Wondering if we should simply relax the spec and indicate that an error should be throw, without declaring a type. Similarly, update the test to check that an error is thrown, without checking its type.